core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -216,21 +229,98 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
-                          request: WorkspaceRequest
-  ): Future[Workspace] = Future.failed(
-    RawlsExceptionWithErrorReport(
-      ErrorReport(StatusCodes.NotImplemented, "WOR-625")
-    )
-  )
+                          request: WorkspaceRequest,
+                          parentContext: RawlsRequestContext
+  ): Future[Workspace] = {
+
+    def createNewWorkspaceRecord(): Future[Workspace] =
+      dataSource.inTransaction { access =>
+        for {
+          _ <- failIfWorkspaceExists(request.toWorkspaceName)
+          workspaceId = UUID.randomUUID()
+          newWorkspace <- createMultiCloudWorkspaceInDatabase(
+            workspaceId.toString,
+            request.toWorkspaceName,
+            request.attributes,
+            access,
+            parentContext
+          )
+        } yield newWorkspace
+      }
+
+    for {
+      // The call to WSM is asynchronous. Before we fire it off, allocate a new workspace record
+      // to avoid naming conflicts - we'll erase it should the clone request to WSM fail.
+      newWorkspace <- createNewWorkspaceRecord()
+
+      cloneResult <- traceWithParent("workspaceManagerDAO.cloneWorkspace", parentContext) { context =>
+        Future(blocking {
+          workspaceManagerDAO.cloneWorkspace(
+            sourceWorkspaceId = sourceWorkspace.workspaceIdAsUUID,
+            workspaceId = newWorkspace.workspaceIdAsUUID,
+            displayName = request.name,
+            spendProfile = profile,
+            context
+          )
+        })
+      }.recoverWith { case t: Throwable =>
+        logger.warn(
+          "Clone workspace request to workspace manager failed for " +
+            s"[ sourceWorkspaceName='${sourceWorkspace.toWorkspaceName}'" +
+            s", newWorkspaceName='${newWorkspace.toWorkspaceName}'" +
+            s"]",
+          t
+        )
+        dataSource.inTransaction(_.workspaceQuery.delete(newWorkspace.toWorkspaceName)) >> Future.failed(t)
+      }
+
+      _ = clonedMultiCloudWorkspaceCounter.inc()
+      _ = logger.info(
+        "Created workspace record and clone job for azure workspace " +
+          s"[ workspaceId='${newWorkspace.workspaceId}'" +
+          s", workspaceName='${newWorkspace.toWorkspaceName}'" +
+          s", cloneJobReportId='${cloneResult.getJobReport.getId}'" +
+          s"]"
+      )
+
+      // We can't specify the jobId to WSM at the time of writing and
+      // a 22-character base64url-encoded short-UUID is generated instead.
+      jobId = WorkspaceManagerDAO.decodeShortUuid(cloneResult.getJobReport.getId).getOrElse {
+        // If this happens we're in trouble. Our database needs a UUID but whatever WSM gave us
+        // was not something we know how to convert into a UUID. I don't think we should delete the
+        // workspace record but I'm not going to write an elaborate error handler to clean this up,
+        // instead favouring PF-1269 as a better solution.
+        val message =
```

Comment:
Don't you need to delete the Rawls workspace record in this case?

Reply:
It's such an unlikely thing to happen that I don't think putting more time into writing and testing this is worth it. If it does happen, it'll be logged and the workspace record will exist so we can try to find out what happened

Reply:
Ha, I totally didn't read your comment in the code! Sure, that's fine.